crates/oxc_formatter/src/lib.rs

```diff
@@ -8,19 +8,39 @@ use oxc_ast::ast::*;
 
 pub use crate::gen::Gen;
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum EndOfLine<'a> {
+    /// Line Feed only (`\n`), common on Linux and macOS as well as inside git repos
+    LF,
+    /// Carriage Return + Line Feed characters (`\r\n`), common on Windows
+    CRLF,
+    /// Carriage Return character only (`\r`), used very rarely
+    CR,
+    /// Maintain existing line endings (mixed values within one file are normalised by looking at what’s used after the first line)
+    Auto(&'a str),
```

Comment:
Does it make sense to just make a tiny allocation for this so we can remove the lifetime?